haddock-api/src/Haddock/Interface/Rename.hs

```diff
@@ -68,20 +68,23 @@ renameInterface dflags renamingEnv warnings iface =
       -- Note that since the renamed AST represents equality constraints as
       -- @HasOpTy t1 eqTyCon_RDR t2@ (and _not_ as @HsEqTy t1 t2@), we need to
       -- manually filter out 'eqTyCon_RDR' (aka @~@).
-      strings = [ pretty dflags n
-                | n <- missingNames
-                , not (isSystemName n)
-                , not (isBuiltInSyntax n)
-                , Exact n /= eqTyCon_RDR
-                ]
+      msgs = [ (n, pretty dflags n)
+             | n <- missingNames
+             , not (isSystemName n)
+             , not (isBuiltInSyntax n)
+             , Exact n /= eqTyCon_RDR
+             ]
+
+      warnMsg (name, msg) =
+        L (nameSrcSpan name) $
+        "Warning: " ++ moduleString (ifaceMod iface) ++
+        ": could not find link destinations for: " ++ msg
```

Comment:
I think this is the biggest issue in this PR, and it is a totally non-obvious one. `nameSrcSpan name` is unfortunately _not_ the location we want. Try it out. We get these names from querying the `tcg_rdr_env` in GHC, so the location attached to the names is pretty much never useful.

There's probably some interesting location to attach to this warning (maybe the location of the initial docstring?) but piping that location through is going to take some care. If you want, you can duck out of this with a plain module location (if you take this path, let's go back to generating just one warning per module).

The warning message shouldn't have `Warning:` or  `moduleString (ifaceMod iface)` though...

Reply:
I see why this name is not useful.  I'm still trying to figure out where I might find the initial docstring's location.  It can't be in the `Interface`, I don't think, because that's where we got the problematic `Name` from in the first place (e.g. `ifaceDocMap`, `ifaceArgMap`).  Should I be seeking a way to attach `SrcSpan` info to the `Maps` in `Haddock.Interface.Create.mkMaps`, using the info in the given `LhsDecl GhcRn`s?

Can you provide any hints off the top of your head?

Reply:
Sorry, I meant the `HsDocString`s corresponding to the `LhsDecl GhcRn`s.

Reply:
Yeah, I even did a bunch of this a while ago. This should serve as a pretty good inspiration: https://github.com/harpocrates/haddock/commit/2b508013bc0fb348c909e6168b5c57ce617f4282 (it's almost exactly what you need). Basically, try to replace as many `HsDocString`'s with `LHsDocString`'s.

Reply:
I'm still struggling to get this running, but I'm getting over my head when it comes to the parser side of things.  I started out when I last posted by changing `HsDocString` to `LHsDocString` everywhere, but it seems like in `LexParseRn.hs` and `Parser.hs` a bunch of individual docstrings get merged into a `MetaDoc` of some kind, so I can't figure out a way to avoid digging in and instead attach location information afterwards.  Your branch makes a lot of the same changes, but it modifies and relies on a type that doesn't seem to exist for me on `ghc-head` (`NsRdrName`) when I get into this part of the code (I'm not sure it compiles; I think there is some missing namespace plumbing as well?).

Having spent all afternoon chugging away at this, I think I'm out of time to keep poking, and I'm going to just report one warning per module and give the module location.

Reply:
That sounds fine to me. I agree that this is a pretty deep rabbit hole too - the one warning per module with the module location is already a step forwards. 😄